src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -121,7 +121,7 @@ - (void)runLoop
 
 + (void)setExpirationForMessage:(TSMessage *)message
 {
-    dispatch_async(self.serialQueue, ^{
+    dispatch_sync(self.serialQueue, ^{
```

Comment:
PTAL @charlesmchen 

I considered adding a completion handler, but the method chain is really deep, across multiple dispatch_asyncs. So I opted for a `sleep` hack in the tests. I really don't like to do this, but I think it's more important we get the tests running.

e.g. since I last picked up this pull request there were 3 or so other changes that broke a bunch of tests. Commits to fix those are also included in this PR.